File: stack/balancingSymbols.cpp
#include <iostream>
#include <string>
#include <vector>

int main()
{
    std::string data;
    std::cout << "Enter the data : " << std::endl;
    std::cin >> data;
    std::vector<int> checkTable;
    int temp,req;
    for (char c : data)
    {
        temp = c;
        if (temp == 91 || temp == 40 || temp == 123)
            checkTable.push_back(temp);
        else if (temp == 93 || temp == 41 || temp == 125)
            {
                if (checkTable.size() == 0)
                {
                    std::cerr << "Error : The entered data is not balanced";
                    exit(1);
                }
                req = (temp == 41) ? temp - 1 : temp - 2;
                if (req != checkTable.back())
                {
                    std::cerr << "Error : The entered data is not balanced";
                    exit(1);
                }   
                checkTable.pop_back();
            }

    }
        
    if (checkTable.size() != 0)
        std::cerr << "Error : the entered data is not balanced ";
    else
        std::cout << "Success : the entered data is balanced";
    
    return 0;
}

File: stack/balancingSymbols.rs
fn main()
{
 let mut input_line = String::new();

 println!("Enter the data : ");
 std::io::stdin().read_line(&mut input_line).expect("Error while reading the data");

 let mut data_stack : Vec<u8>= Vec::new();
 let mut av : u8;
 let mut req : u8;
 for val in input_line.chars()
 {
  av = val as u8;
  if av == 91 || av == 40 || av == 123
   {data_stack.push(av);}
  if av == 93 || av == 41 || av == 125
   {
    if data_stack.len() == 0
     {
      println!("Error : The given string is not balanced ");
      std::process::exit(0);
     }
    req = if av == 41 {av - 1} else {av-2};
    if data_stack[data_stack.len()-1] != req
     {
      println!("Error : The given string is not balanced ");
      std::process::exit(0);
     }
    data_stack.pop();
   }   
 }
 if data_stack.len() != 0
 {
  println!("Error : The given string is not balanced ");
 }
 else
 {
  println!("Success : the given string is balanced ");
 }

}